commands/roles.js

```diff
@@ -0,0 +1,22 @@
+const { roleBlacklist } = require('../config.json');
+
+module.exports = {
+	name: 'roles',
+	description: 'Lists all self-assignable roles.',
+	aliases: [],
+	usage: '[roles]',
+	cooldown: 5,
+	guildOnly: true,
+	execute(message, args) {
+
+        let assignableRoles = message.guild.roles.values()
+            .filter(role=>!roleBlacklist.includes(role.name))
+            .map(role=>role.name);
+
+        let assignableRoleNames = assignableRoles.reduce((a,b)=>`\`${a}\`, \`${b}\``);
```

Comment:
But we could filter here instead, being mindful that `@everyone` is the default role so we don't need it in our list. And sort it alphabetically, so it scans well, too. 🤭

```js
let assignableRoles = roles.filter(role => !roleBlacklist.includes(role)).slice(1).sort();
let assignableRoleList = `\`${assignableRoles.join("`, `")}\``;
```